src/Compilers/Core/Portable/EncodedStringText.cs

```diff
@@ -16,18 +16,36 @@ internal static class EncodedStringText
         /// Encoding to use when there is no byte order mark (BOM) on the stream. This encoder may throw a <see cref="DecoderFallbackException"/>
         /// if the stream contains invalid UTF-8 bytes.
         /// </summary>
-        private static readonly Encoding s_fallbackEncoding = new UTF8Encoding(encoderShouldEmitUTF8Identifier: false, throwOnInvalidBytes: true);
+        private static readonly Encoding s_utf8Encoding = new UTF8Encoding(encoderShouldEmitUTF8Identifier: false, throwOnInvalidBytes: true);
 
         /// <summary>
-        /// Encoding to use when UTF-8 fails and default OS ANSI encoding is not available. If available, we use CodePage 1252. If not, we use Latin1.
+        /// Encoding to use when UTF-8 fails. We try to find the following, in order, if available:
+        ///     1. The default ANSI codepage
+       ///      2. CodePage 1252.
+       ///      3. Latin1.
         /// </summary>
-        private static readonly Encoding s_defaultFallbackEncoding = GetDefaultFallbackEncoding();
+        private static readonly Encoding s_fallbackEncoding = GetFallbackEncoding();
 
-        private static Encoding GetDefaultFallbackEncoding()
+        private static Encoding GetFallbackEncoding()
         {
             try
             {
-                return PortableShim.Encoding.GetEncoding(1252);
+                // If we're running on CoreCLR there is no "default" codepage
+                // but we should be able to grab 1252 from System.Text.Encoding.CodePages
+                if (CoreClrShim.IsCoreClr)
+                {
+                    CoreClrShim.Encoding.RegisterProvider(CoreClrShim.CodePagesEncodingProvider.Instance);
+                    // We should now have 1252 from the CodePagesEncodingProvider
+                    return PortableShim.Encoding.GetEncoding(1252);
```

Comment:
I thought it would work if we register CodePagesEncodingProvider prior to the call, which we do. We need to confirm whether this is the case.